### ISSUE TITLE
Temperature not setting v3.1.2

### ISSUE THREAD

**Issue by Olegaus** [jeffschubert/homebridge-daikin-oneplus#30]
Temperature not setting since updating to v3.1.2   Requested token, reinstalled, restarted bridge/hoobs more then once. downgraded to v 3.1.1. everthing works again. updated to 3.1.2 same issue. have no choice but to downgrade to 3.1.1. to be able to  control Daikin one plus. Note: Daiking native app support show software version 3.1.4 as of Jan 21/23, also not sure if it makes a difference but native Daikin app support show cloud services set to monitor, however it works fine on v3.1.1. Please advise.

**Comment by jeffschubert** [comment]
Can you give me more details about the mode and such your thermostat is set to? Mine is on Heat and I can change the temp without issue at the moment. My thermostat is at version 3.1.4 as well. 
Oh and what version of Homebridge are you on?

**Comment by Olegaus** [comment]
Mode set to auto as usual. when changing temp in HOMEKIT OR HOOBS after a few seconds it goes back to what it was before. Hoobs is up to date on latest firmware 1.5.0. logs looked OK EXCEPT TODAY I noticed some errors see below:

1/21/2023, 11:57:30 AMDAIKIN+ BridgeDaikin One+Checking for Circulate Air Fan...
1/21/2023, 12:46:26 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
1/21/2023, 12:46:26 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
1/21/2023, 12:46:26 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
1/21/2023, 12:46:26 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
1/21/2023, 7:06:05 PMDAIKIN+ BridgeDaikin One+ERRORError getting token: []
1/21/2023, 7:06:05 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
1/21/2023, 7:06:05 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
1/21/2023, 10:33:06 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
1/21/2023, 10:33:06 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
1/21/2023, 10:33:06 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
1/21/2023, 10:33:06 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
1/22/2023, 3:06:57 AMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
1/22/2023, 3:06:57 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
1/22/2023, 3:06:57 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
1/22/2023, 3:06:57 AMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []

Previously tried to reset all tokens in native Daikin one app and requesting new one. same result

**Comment by Olegaus** [comment]
updated to 3.1.2. restarted/reset token/requested new one/restarted. rebooted daikin too. new log errors:

1/22/2023, 11:40:13 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Current Heating Cooling State] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'equipmentStatus')
1/22/2023, 11:40:13 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Heating Threshold Temperature] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'hspActive')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Target Heating Cooling State] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'mode')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Target Temperature] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'mode')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Cooling Threshold Temperature] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'cspActive')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Temperature Display Units] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'units')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Target Relative Humidity] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'humSP')
1/22/2023, 11:40:14 AMDAIKIN+ BridgeERROR[DAIKIN+ Bridge A90E@Main Room Thermostat@@Current Temperature] Unhandled error thrown inside read handler for characteristic: Cannot read properties of undefined (reading 'tempIndoor')

**Comment by Olegaus** [comment]
Update. V3.1.2 switching modes (OFF/heat/cooling/auto) works. Setting temp is still not working.

**Comment by jeffschubert** [comment]
Thanks for the logs. It doesn't seem to be getting any data from the Daikin api that I can tell. Which makes it odd that changing modes works. Hopefully I'll have some spare time an evening this week to dig into it deeper. 
If you're using hoobs, I'm not familiar with how it differs from homebridge. I've never tested with hoobs so there could be a deeper package incompatibility somehow. I'm on homebridge 1.6.0 and I believe there were some significant lower level changes in that version. If hoobs hasn't been updated accordingly that could be some of it. 
I'll check further when I have time.

**Comment by Olegaus** [comment]
Thank you.

**Comment by jeffschubert** [comment]
@Olegaus Are you still having issues setting the temp? If it successfully switches modes, then it is successfully connected to the Daikin API. It should thus also be able to change the temp.
But if only switching modes works, can you send logs again of switching modes and then trying to set the temp? That will hopefully point me in the right direction.

**Comment by Olegaus** [comment]
Updated to 3.1.2 restarted bridge/restarted network router/restarted home bridge. Mode switching work, temp setting is not. Nothing abnormal in the logs.. Downgraded to 3.1.1 again,  no problem, everything works again. Don't know what else to try. Thank you. Will stay on 3.1.1 for now.

**Comment by SPJONM3** [comment]
I’m having the same issue. I can’t sent temp on my One+ when using v3.1.2. I reverted to 3.1.1 and setting temp appears to be working again. I did not need to request a new token from Daikin.

**Comment by jeffschubert** [comment]
@SPJONM3 Are you on Hoobs as well? If not, what version of Homebridge?

I checked again and setting the temp works for me on Homebridge 1.6.0 and 3.1.2 of the plugin.

**Comment by SPJONM3** [comment]
@jeffschubert I’m on HB 1.6.0, Node.js 18.13.0

**Comment by jeffschubert** [comment]
@SPJONM3 Ahh, I'm on Node.js 16.16.0. Perhaps that's the key. Let me investigate further.

**Comment by SPJONM3** [comment]
Let me know if you need anymore details.

**Comment by jeffschubert** [comment]
Thanks. I was able to update my dev homebridge to node 18.13.0 and setting the temp works. This is quite odd. Where is your Homebridge installed? Raspberry Pi, Synology, Docker, etc?

I think my next step will be to add a toggle to enable debug logging so it will spit out more details. Unfortunately, I don't have time today, but hopefully within the next few days/week. 

For now, I'd recommend going back to 3.1.1 until then.


If you're comfortable with the command line and familiar enough with your homebridge installation, you can stop homebridge and then run it from the terminal with a `-D` argument like so: `homebridge -D`. (If that doesn't make sense, just ignore and wait for a new version of the plugin with debug log capabilities.)

That'll cause it to spit out debug logs for homebridge and any plugins including mine. Then try setting the temp and post any logs you see. When you use the Apple Home app to set the temp, the debug logs will show something like the following. I'd expect to always see the first line and then an error or something different with the other lines. The device id (ac23...) will definitely be different.

```
[1/28/2023, 9:42:45 PM] [Daikin One+] Thermostat - Set TargetTemperature: 20.5
[1/28/2023, 9:42:45 PM] [Daikin One+] setTargetTemps-> device: ac230f4c-d900-11ea-b7e2-9bb9e77f74dd; requestData: {"hspHome":20.5} []
[1/28/2023, 9:42:46 PM] [Daikin One+] setTargetTemps-> device: ac230f4c-d900-11ea-b7e2-9bb9e77f74dd; response:{"message":"Write sent"} []
[1/28/2023, 9:42:46 PM] [Daikin One+] Updated cache for ac230f4c-d900-11ea-b7e2-9bb9e77f74dd - {"hspHome":20.5} []
```

**Comment by SPJONM3** [comment]
It’s installed on Debian 11 x64, which is a VM running on an ESXi server. 

If I have time tomorrow, I’ll try running HB with the -D argument.

**Comment by Olegaus** [comment]
Here is some logs running on 3.1.1 hope it helps
2/2023, 2:54:07 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/2/2023, 2:54:07 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 2:54:07 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 3:16:08 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/2/2023, 3:16:08 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 3:16:08 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 4:47:16 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/2/2023, 4:47:16 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 4:47:16 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 7:26:35 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 7:26:35 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/2/2023, 7:26:35 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 10:49:56 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 10:49:56 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 10:49:56 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/2/2023, 10:49:56 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/2/2023, 11:01:58 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/2/2023, 11:01:58 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/2/2023, 11:01:58 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/2/2023, 11:01:58 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/3/2023, 12:28:55 AMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/3/2023, 12:28:55 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 12:28:55 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 4:01:54 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 4:01:54 AMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/3/2023, 4:01:54 AMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/3/2023, 4:01:54 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 4:55:55 AMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/3/2023, 4:55:55 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 4:55:55 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 7:36:52 AMDAIKIN+ BridgeDaikin One+ERRORError getting token: []
2/3/2023, 7:36:52 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 7:36:52 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 11:27:57 AMDAIKIN+ BridgeDaikin One+ERRORError getting token: []
2/3/2023, 11:27:57 AMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 11:27:57 AMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 2:34:34 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/3/2023, 2:34:34 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 2:34:34 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 2:34:34 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/3/2023, 4:16:22 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 4:16:22 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 4:16:22 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/3/2023, 5:25:19 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/3/2023, 5:25:19 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/3/2023, 5:25:19 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 5:25:19 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 6:47:40 PMDAIKIN+ BridgeDaikin One+ERRORError with getRequest: https://api.daikinskyport.com/deviceData/41c687a2-b666-11eb-bbff-b36d53af0026 []
2/3/2023, 6:47:40 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []
2/3/2023, 6:47:40 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 6:47:40 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 9:31:20 PMDAIKIN+ BridgeDaikin One+ERROR[object Object] []
2/3/2023, 9:31:20 PMDAIKIN+ BridgeDaikin One+ERRORError with request: []
2/3/2023, 9:31:20 PMDAIKIN+ BridgeDaikin One+ERRORError refreshing token: []
2/3/2023, 11:19:37 PMDAIKIN+ BridgeDaikin One+ERRORUnable to retrieve data for Main Room. []

**Comment by Olegaus** [comment]
Updated to 3.1.2 in the logs says plug-in site not available 
2/4/2023, 4:21:00 PMDAIKIN+ BridgeDaikin One+Checking for Emergency Heat Switch...
2/4/2023, 4:21:00 PMDAIKIN+ BridgeDaikin One+Checking for One Clean Fan...
2/4/2023, 4:21:00 PMDAIKIN+ BridgeDaikin One+Checking for Circulate Air Fan...
2/4/2023, 4:24:06 PMWARNINGplugin site unavailable

**Comment by cdumez** [comment]
For what it's worth, setting the temperature has also stopped working for me. I am running Homebridge v1.6.0 on a Raspberry Pi with homebridge-daikin-oneplus v3.1.2 plugin.

I see the following error in the logs which may be related:
```
[26/02/2023, 03:19:25] [Daikin One+] Error refreshing token: []
[26/02/2023, 03:19:25] [Daikin One+] Error with request: []
[26/02/2023, 03:19:25] [Daikin One+] <ref *2> Writable {
  _writableState: WritableState {
    objectMode: false,
    highWaterMark: 16384,
    finalCalled: false,
    needDrain: false,
    ending: false,
    ended: false,
    finished: false,
    destroyed: false,
    decodeStrings: true,
    defaultEncoding: 'utf8',
    length: 0,
    writing: false,
    corked: 0,
    sync: true,
    bufferProcessing: false,
    onwrite: [Function: bound onwrite],
    writecb: null,
    writelen: 0,
    afterWriteTickInfo: null,
    buffered: [],
    bufferedIndex: 0,
    allBuffers: true,
    allNoop: true,
    pendingcb: 0,
    constructed: true,
    prefinished: false,
    errorEmitted: false,
    emitClose: true,
    autoDestroy: true,
    errored: null,
    closed: false,
    closeEmitted: false,
    [Symbol(kOnFinished)]: []
  },
  _events: [Object: null prototype] {
    response: [Function: handleResponse],
    error: [Function: handleRequestError],
    socket: [Function: handleRequestSocket]
  },
  _eventsCount: 3,
  _maxListeners: undefined,
  _options: {
    maxRedirects: 21,
    maxBodyLength: 10485760,
    protocol: 'https:',
    path: '/users/auth/token',
    method: 'POST',
    headers: {
      Accept: 'application/json',
      'Content-Type': 'application/json',
      'User-Agent': 'axios/0.27.2',
      'Content-Length': 1827
    },
    agent: undefined,
    agents: { http: undefined, https: undefined },
    auth: undefined,
    hostname: 'api.daikinskyport.com',
    port: null,
    nativeProtocols: { 'http:': [Object], 'https:': [Object] },
    pathname: '/users/auth/token'
  },
  _ended: false,
  _ending: true,
  _redirectCount: 0,
  _redirects: [],
  _requestBodyLength: 1827,
  _requestBodyBuffers: [
    {
      data: <Buffer 7b 22 65 6d 61 69 6c 22 3a 22 63 68 72 69 73 40 64 75 6d 65 7a 2e 6f 72 67 22 2c 22 72 65 66 72 65 73 68 54 6f 6b 65 6e 22 3a 22 65 79 4a 6a 64 48 6b ... 1777 more bytes>,
      encoding: undefined
    }
  ],
  _onNativeResponse: [Function (anonymous)],
  _currentRequest: <ref *1> ClientRequest {
    _events: [Object: null prototype] {
      response: [Function],
      abort: [Function (anonymous)],
      aborted: [Function (anonymous)],
      connect: [Function (anonymous)],
      error: [Function (anonymous)],
      socket: [Function (anonymous)],
      timeout: [Function (anonymous)]
    },
    _eventsCount: 7,
    _maxListeners: undefined,
    outputData: [],
    outputSize: 0,
    writable: true,
    destroyed: false,
    _last: true,
    chunkedEncoding: false,
    shouldKeepAlive: false,
    maxRequestsOnConnectionReached: false,
    _defaultKeepAlive: true,
    useChunkedEncodingByDefault: true,
    sendDate: false,
    _removedConnection: false,
    _removedContLen: false,
    _removedTE: false,
    strictContentLength: false,
    _contentLength: 1827,
    _hasBody: true,
    _trailer: '',
    finished: false,
    _headerSent: true,
    _closed: false,
    socket: TLSSocket {
      _tlsOptions: [Object],
      _secureEstablished: false,
      _securePending: false,
      _newSessionPending: false,
      _controlReleased: true,
      secureConnecting: true,
      _SNICallback: null,
      servername: null,
      alpnProtocol: null,
      authorized: false,
      authorizationError: null,
      encrypted: true,
      _events: [Object: null prototype],
      _eventsCount: 11,
      connecting: false,
      _hadError: true,
      _parent: null,
      _host: 'api.daikinskyport.com',
      _closeAfterHandlingError: false,
      _readableState: [ReadableState],
      _maxListeners: undefined,
      _writableState: [WritableState],
      allowHalfOpen: false,
      _sockname: null,
      _pendingData: [Array],
      _pendingEncoding: '',
      server: undefined,
      _server: null,
      ssl: null,
      _requestCert: true,
      _rejectUnauthorized: true,
      parser: null,
      _httpMessage: [Circular *1],
      [Symbol(res)]: [TLSWrap],
      [Symbol(verified)]: false,
      [Symbol(pendingSession)]: null,
      [Symbol(async_id_symbol)]: 32418,
      [Symbol(kHandle)]: null,
      [Symbol(lastWriteQueueSize)]: 0,
      [Symbol(timeout)]: null,
      [Symbol(kBuffer)]: null,
      [Symbol(kBufferCb)]: null,
      [Symbol(kBufferGen)]: null,
      [Symbol(kCapture)]: false,
      [Symbol(kSetNoDelay)]: false,
      [Symbol(kSetKeepAlive)]: true,
      [Symbol(kSetKeepAliveInitialDelay)]: 60,
      [Symbol(kBytesRead)]: 0,
      [Symbol(kBytesWritten)]: 0,
      [Symbol(connect-options)]: [Object]
    },
    _header: 'POST /users/auth/token HTTP/1.1\r\n' +
      'Accept: application/json\r\n' +
      'Content-Type: application/json\r\n' +
      'User-Agent: axios/0.27.2\r\n' +
      'Content-Length: 1827\r\n' +
      'Host: api.daikinskyport.com\r\n' +
      'Connection: close\r\n' +
      '\r\n',
    _keepAliveTimeout: 0,
    _onPendingData: [Function: nop],
    agent: Agent {
      _events: [Object: null prototype],
      _eventsCount: 2,
      _maxListeners: undefined,
      defaultPort: 443,
      protocol: 'https:',
      options: [Object: null prototype],
      requests: [Object: null prototype] {},
      sockets: [Object: null prototype],
      freeSockets: [Object: null prototype] {},
      keepAliveMsecs: 1000,
      keepAlive: false,
      maxSockets: Infinity,
      maxFreeSockets: 256,
      scheduling: 'lifo',
      maxTotalSockets: Infinity,
      totalSocketCount: 1,
      maxCachedSessions: 100,
      _sessionCache: [Object],
      [Symbol(kCapture)]: false
    },
    socketPath: undefined,
    method: 'POST',
    maxHeaderSize: undefined,
    insecureHTTPParser: undefined,
    path: '/users/auth/token',
    _ended: false,
    res: null,
    aborted: false,
    timeoutCb: null,
    upgradeOrConnect: false,
    parser: null,
    maxHeadersCount: null,
    reusedSocket: false,
    host: 'api.daikinskyport.com',
    protocol: 'https:',
    _redirectable: [Circular *2],
    [Symbol(kCapture)]: false,
    [Symbol(kBytesWritten)]: 0,
    [Symbol(kEndCalled)]: false,
    [Symbol(kNeedDrain)]: false,
    [Symbol(corked)]: 0,
    [Symbol(kOutHeaders)]: [Object: null prototype] {
      accept: [Array],
      'content-type': [Array],
      'user-agent': [Array],
      'content-length': [Array],
      host: [Array]
    },
    [Symbol(errored)]: null,
    [Symbol(kUniqueHeaders)]: null
  },
  _currentUrl: 'https://api.daikinskyport.com/users/auth/token',
  [Symbol(kCapture)]: false
} []
```

**Comment by jeffschubert** [comment]
@cdumez Thanks for the log. It unfortunately doesn't tell me much. I continue to have no issues setting the temp on my installation. Work and life have prevented me from spending more time on this recently. But I hope to get a debug flag implemented to make it easier to get more details.

**Comment by Rekishi2002** [comment]
I have experienced the same issue others are calling out, although I have not back-revved to 3.1.1 to see if the ability to set the temp returns.

I find that if I used the Daikin One app, I am able to adjust the temperature, and it immediately takes effect on the thermostat. Conversely, when adjust via HomeKit, the UI indicates that the changes have been made, until the tile is closed.

At which point the temps fall back to what they were previously.

If you are not at the thermostat, this might not be noticed.

I doubt it will be much help, but I do work for Daikin and know the engineers who worked to get this thermostat out. However, I do not know who the software developers are, but I can ask. I believe that the backend is not specific to this thermostat, but is rather a global cloud service (I could be wrong).

Separate topic - but I have had issues with keeping the thermostat connected to WiFi at times. It *seems* like it may be related to my Eero mesh trying to shift control between different devices. I have the Pro 6E (3), and added them to a previous 5 Pro (5) network. I think that there were too many shifts between devices possibly (the option to auto assign is turned on - and I need it on for things like my Apple Watch).

I removed the two 5 Pro devices, and saw a significant drop in WiFi connection drop on the thermostat....but it is potentially anecdotal. 

Anyone else seeing similar issues on Eero and the One+ thermostat?

**Comment by jonnyborbs** [comment]
I have the same behavior as @Rekishi2002 - changing settings in the One app reflects those changes instantly in Homebridge/HomeKit. Changing anything in HomeKit directly looks like it took effect until the tile is closed, then it reverts back with no changes made or errors surfaced.

**Comment by jeffschubert** [comment]
Sorry for the slow response. I've made some changes for the next release that should fix this regression in 3.1.2. I don't think I have every edge case solved yet, but I'm chipping away. Thanks for your patience.

**Comment by jonnyborbs** [comment]
No worries at all! The prior version is working properly less the enhancements added in 3.1.2, and I'm sure we all appreciate you building this in the first place. Thanks for looking into it!

**Comment by jonnyborbs** [comment]
Hiya @jeffschubert - was just bumping this one up 🙂

**Comment by jeffschubert** [comment]
This should be fixed in the new 3.2.0 version, but let me know if you're still having issues.

**Comment by jonnyborbs** [comment]
Just upgraded and tested out temperature setting back and forth between the One app and HomeKit and things seem to be behaving as expected in both directions. HomeKit is a little slow to reflect changes made directly on the thermostat or in One, in fact I had to close and reopen the Home app to see them - not sure if this is the plugin needing to send more frequent updates or something on the Apple side, but can confirm the issue I was seeing in 3.1.2 is fixed! Thanks so much for the effort here.
